src/guides/v2.3/frontend-dev-guide/templates/template-email-migration.md

```diff
@@ -0,0 +1,230 @@
+In this example, we will create and pass a `lifetime_spend` custom value.
+
+1. Create a class that implements `Magento\Framework\Filter\SimpleDirective\ProcessorInterface`:
+
+   ```php
+    declare(strict_types=1);
+    namespace GadgetCorp\CustomEmailDirective\Model;
+    use Magento\Framework\Filter\SimpleDirective\ProcessorInterface;
+    use Magento\Framework\Pricing\PriceCurrencyInterface;
+    /**
+    * Calculates the lifetime spend of all customers
+    */
+    class LifetimeSpendDirective implements ProcessorInterface
+    {
+        /**
+        * @var PriceCurrencyInterface
+        */
+        private $priceCurrency;
+        /**
+        * @param PriceCurrencyInterface $priceCurrency
+        */
+        public function __construct(PriceCurrencyInterface $priceCurrency)
+        {
+            $this->priceCurrency = $priceCurrency;
+        }
+        /**
+        * @inheritDoc
+        */
+        public function getName(): string
+        {
+            return 'lifetime_spend';
+        }
+        /**
+        * @inheritDoc
+        */
+        public function process($value, array $parameters, ?string $html): string
+        {
+            $shouldBold = !empty($parameters['should_bold']);
+            $amount = $this->priceCurrency->getCurrencySymbol() . $this->calculateLifetimeSpend();
+            return ($shouldBold ? '<strong>' . $amount . '</strong>' : $amount);
+        }
+        /**
+        * @inheritDoc
+        */
+        public function getDefaultFilters(): ?array
+        {
+            // Make sure newlines are converted to <br /> tags by default
+            return ['nl2br'];
+        }
+        /**
+        * Calculate the total amount of money spent by all customers for all time
+        *
+        * @return float
+        */
+        private function calculateLifetimeSpend(): float
+        {
+            // Add code here to calculate the lifetime spend
+            return 123.45;
+        }
+    }
+   ```
+
+   and save the file to <Vendor>/<module>/Model.
+
+1. Add the new directive to the pool by adding this block to `di.xml`.
+
+   ```xml
+   <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
+    <type name="Magento\Framework\Filter\SimpleDirective\ProcessorPool">
+        <arguments>
+            <argument name="processors" xsi:type="array">
+                <item name="lifetime_spend" xsi:type="object">GadgetCorp\CustomEmailDirective\Model\LifetimeSpendDirective</item>
+            </argument>
+        </arguments>
+    </type>
+   </config>
+   ```
+
+The new variable is now available within the email template as `{{lifetime_spend}}`.
+Note in the class above, we also defined the filter `shouldBold`. We can use that with `{{lifetime_spend should_bold=1}}`.
+You may also use multiple filters within a var statement: `{{lifetime_spend should_bold=1 |escape}}`.
```

Comment:
As mentioned, `should_bold` is a parameter so multiple filters would look like this:
```
You may also use multiple filters within a var statement: `{{lifetime_spend should_bold=1 |escape|nl2br}}`.
```